common.lic

```diff
@@ -382,6 +390,19 @@ module DRC
     end
   end
 
+  # Looks up the canonical name of the town based on the given text.
+  # Utility to help identify the canonical town name based on arbitrary text.
+  # For example, "Theren" for "Therenborough" and "Haven" for "Riverhaven".
+  # It also handles missing apostrophes and the occasional space between names
+  # like "merkresh" or "Mer'Kresh" or "ainghazal" or "Ain Ghazal".
+  # Returns nil if unable to find a match.
+  def get_town_name(text)
+    $HOMETOWN_REGEX
+      .select { |town, regex| regex =~ text }
+      .keys
+      .first
```

Comment:
I'm wondering about this `.first` call. Is it a problem if `text` matches more than 1 town?

Should we notify the user we're choosing the first one but they should be more explicit?

Reply:
Good question. The regexes _should not_ (in my opinion) allow for ambiguous results. There should be exactly 0 or 1 town name found.

But, the day that a regex were to be added/updated that was too loose could cause multiples to be found (as you point out) and we should protect the user from unexpected results.

At the moment I'm leaning towards the script should error that prints the results and asks the user to use a more specific town value and to contact the devs/raise an issue about the ambiguity.